internal/provider/resource/project_template_resource.go

```diff
@@ -0,0 +1,676 @@
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &ProjectTemplateResource{}
+)
+
+type ProjectTemplateResource struct {
+	client *infisical.Client
+}
+
+type PermissionModel struct {
+	Subject   types.String `tfsdk:"subject"`
+	Action    types.Set    `tfsdk:"action"`
+	Condition types.String `tfsdk:"condition"`
+	Inverted  types.Bool   `tfsdk:"inverted"`
+}
+
+type RoleModel struct {
+	Name        types.String `tfsdk:"name"`
+	Slug        types.String `tfsdk:"slug"`
+	Permissions types.List   `tfsdk:"permissions"`
+}
+
+type EnvironmentModel struct {
+	Name     types.String `tfsdk:"name"`
+	Slug     types.String `tfsdk:"slug"`
+	Position types.Int64  `tfsdk:"position"`
+}
+
+type ProjectTemplateResourceModel struct {
+	ID           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Description  types.String `tfsdk:"description"`
+	Roles        types.List   `tfsdk:"roles"`
+	Environments types.List   `tfsdk:"environments"`
+	Type         types.String `tfsdk:"type"`
+}
+
+func NewProjectTemplateResource() resource.Resource {
+	return &ProjectTemplateResource{}
+}
+
+// Metadata returns the resource type name.
+func (r *ProjectTemplateResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_project_template"
+}
+
+func (r *ProjectTemplateResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Create project templates & save to Infisical. Only Machine Identity authentication is supported for this resource.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the project template",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the project template",
+				Required:    true,
+			},
+			"description": schema.StringAttribute{
+				Description: "The description of the project template",
+				Optional:    true,
+			},
+			"type": schema.StringAttribute{
+				Description: "The type of the project template",
```

Comment:
what are the possible values?

let's ensure that the doc is as complete as possible to other users